src/components/custom-field-input-single-choice/custom-field-input-single-choice.jsx

```diff
@@ -41,35 +62,41 @@ export default function CustomFieldInputSingleChoice(props) {
         placeholder={props.placeholder}
         readOnly={props.readOnly}
         required={props.required}
-        value={props.value}
+        value={value.label}
       />
       { showOptions && (
         <Listbox
           className={styles.dropdown}
           labelledBy={`${props.id}-label`}
+          onSelectionChange={onSelectionChange}
           refs={refs}
         >
-          { props.choices.map(item => <ListOption key={item}>{item}</ListOption>) }
+          { listOptions }
         </Listbox>
       ) }
     </div>
   );
 }
 
+const ChoiceType = PropTypes.shape({
```

Comment:
👍  this is great! Do we want `id` and `label` to be required?

Reply:
Doing so caused failures because the default has `id` and `label` as `undefined`, since it's an empty object.